src/code/z_play.c

```diff
@@ -110,34 +110,254 @@
+
+    return globalCtx->roomCtx.currRoom.mesh->type0.type != 1;
+}
+
+s32 FrameAdvance_IsEnabled(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    return globalCtx->frameAdvCtx.enabled != 0;
+}
+
+// Unused, unchanged from OoT, which uses it only in one Camera function.
+/**
+ * @brief Tests if \p actor is a door and the sides are different rooms.
+ *
+ * @param[in] gameState GameState, promoted to globalCtx inside.
+ * @param[in] actor Actor to test.
+ * @param[out] yaw Facing angle of the actor, or reverse if in the back room.
+ * @return true if \p actor is a door and the sides are in different rooms, false otherwise
+ */
+s32 func_8016A02C(GameState* gameState, Actor* actor, s16* yaw) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    TransitionActorEntry* transitionActor;
+    s8 frontRoom;
+
+    if (actor->category != ACTORCAT_DOOR) {
+        return false;
+    }
+
+    transitionActor = &globalCtx->doorCtx.transitionActorList[(u16)actor->params >> 10];
+    frontRoom = transitionActor->sides[0].room;
+    if (frontRoom == transitionActor->sides[1].room) {
+        return false;
+    }
+
+    if (frontRoom == actor->room) {
+        *yaw = actor->shape.rot.y;
+    } else {
+        *yaw = actor->shape.rot.y + 0x8000;
+    }
+
+    return true;
+}
+
+// Unused
+/**
+ * @brief Tests if \p pos is underwater.
+ *
+ * @param[in] globalCtx GlobalContext
+ * @param[in] pos position to test
+ * @return True if inside a waterbox and not above a void.
+ */
+s32 Play_IsUnderwater(GlobalContext* globalCtx, Vec3f* pos) {
+    WaterBox* waterBox;
+    CollisionPoly* poly;
+    Vec3f waterSurfacePos;
+    s32 bgId;
+
+    waterSurfacePos = *pos;
+
+    // == true required to match.
+    if ((WaterBox_GetSurface1(globalCtx, &globalCtx->colCtx, waterSurfacePos.x, waterSurfacePos.z, &waterSurfacePos.y,
+                              &waterBox) == true) &&
+        (pos->y < waterSurfacePos.y) &&
+        (BgCheck_EntityRaycastFloor3(&globalCtx->colCtx, &poly, &bgId, &waterSurfacePos) != BGCHECK_Y_MIN)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+// z_demo and EnTest4
+// This data appears to be a boolean. It is only set by Play_Init.
+s32 func_8016A168(void) {
+    return D_801D0D50;
+}
+
+// List of cutscene numbers.
+extern s16 D_801D0D64[];
+// s16 D_801D0D64[] = { 0xFFFD, 0xFFFE, 0xFFFC, 0xFFFB, 0xFFF9, 0xFFF5, 0xFFF8, 0xFFF7, 0xFFFA, 0xFFF0 };
+// s16 D_801D0D64[] = { -3, -2, -4, -5, -7, -11, -8, -9, -6, -16 };
+
+// Used by Player
+/**
+ * Sets the cutscene numbers in globalCtx->unk_1879C.
+ * Set to -1 by default. If there is an ActorCutscene where unk4 matches the appropriate element of D_801D0D64 (and
+ * possibly change its priority for the zeroth one)
+ */
+void func_8016A178(GameState* gameState, s32 cutscene) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    s32 i;
+    s16* phi_s3 = globalCtx->unk_1879C;
```

Comment:
```suggestion
    s16* phi_s3 = globalCtx->playerActorCsId;
```
I feel comfortable that this is specifically the actorCsIds that are associated with player. Although player and actor together looks a little funny. You can choose to name this here or save it for later